(IL-846) Update Action Button Activity

alexmason-rw opened the pull request:
- Update to Kotlin
- Syntax and style fixes

alexmason-rw committed:
Update ActionButtonActivity

- Update to Kotlin
- Syntax and style fixes

alexmason-rw committed:
Merge branch 'IL-846' into IL-846-ActionButton

alexmason-rw committed:
Update action_button_main.xml

alexmason-rw committed:
Merge branch 'IL-846' into IL-846-ActionButton

AllisonRealwear commented on hmt1developerexamples/src/main/java/com/realwear/hmt1developerexamples/ActionButtonActivity.kt:
```diff
@@ -0,0 +1,90 @@
+ * Activity that shows how to interact with the action button on a HMT-1 device
+ */
+class ActionButtonActivity : Activity() {
+    private val _actionButtonImageView: ImageView by lazy {
```
Have you thought about using [view binding](https://developer.android.com/topic/libraries/view-binding) here?

BBaleilevuka replied:
JETPACK!!! This makes me think jetpack compose might be something we could leverage with our WearHF system to improve performance.

However that's a conversation for another time but this reminded me of it.

BBaleilevuka commented on hmt1developerexamples/src/main/java/com/realwear/hmt1developerexamples/ActionButtonActivity.kt:
```diff
@@ -0,0 +1,90 @@
+        finish()
+    }
+
+    /**
+     * Called when a key is pressed - See Android docs
+     *
+     * @param keyCode The value in event.getKeyCode()
+     * @param event   Description of the key event
+     * @return true if the action button was pressed, otherwise false
+     */
```
For Kotlin code comments there is a style that we have been loosely following: https://kotlinlang.org/docs/coding-conventions.html#documentation-comments

However, Kotlin code comments start with `Returns ...` then provides a briefs explanation. 

This might be a little over-kill for your current project since it's a demo app but I thought it'd be a good style to keep in mind.

alexmason-rw replied:
Oh sure, I can go ahead and convert these comments as well.

BBaleilevuka replied:
Oh I just realized this was the code comments from the original Java docs and it's nothing new. It's up to you but I don't think it's blocking for this PR.

alexmason-rw committed:
View binding, comment formatting, and smaller functions

- Adding in View Binding
- Breaking apart overly verbose onCreate in MainActivity
- Proper comment formatting

BBaleilevuka reviewed (approved):
Looks good to me 👍
